server/plugin.go

```diff
@@ -225,6 +229,109 @@ func (p *Plugin) handleAdd(w http.ResponseWriter, r *http.Request) {
 	p.postReplyIfNeeded(addRequest.PostID, replyMessage, addRequest.Message)
 }
 
+type updateAPIRequest struct {
+	Message string `json:"message"`
+	SendTo  string `json:"send_to"`
+	PostID  string `json:"post_id"`
+}
+
+
+func (p *Plugin) handleUpdate(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	var updateRequest *updateAPIRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&updateRequest)
+	if err != nil {
+		p.API.LogError("Unable to decode JSON err=" + err.Error())
+		p.handleErrorWithCode(w, http.StatusBadRequest, "Unable to decode JSON", err)
+		return
+	}
+
+	senderName := p.listManager.GetUserName(userID)
+
+	if updateRequest.SendTo == "" {
+		_, err = p.listManager.UpdateIssue(userID, updateRequest.Message, updateRequest.PostID)
+		if err != nil {
+			p.API.LogError("Unable to add issue err=" + err.Error())
+			p.handleErrorWithCode(w, http.StatusInternalServerError, "Unable to add issue", err)
+			return
+		}
+		p.trackUpdateIssue(userID, sourceWebapp, updateRequest.PostID != "")
+		replyMessage := fmt.Sprintf("@%s attached a todo to this thread", senderName)
+		p.postReplyIfNeeded(updateRequest.PostID, replyMessage, updateRequest.Message)
+		return
+	}
+
+	receiver, appErr := p.API.GetUserByUsername(updateRequest.SendTo)
+	if appErr != nil {
+		p.API.LogError("username not valid, err=" + appErr.Error())
+		p.handleErrorWithCode(w, http.StatusInternalServerError, "Unable to find user", err)
+		return
+	}
+
+	if receiver.Id == userID {
+		_, err = p.listManager.UpdateIssue(userID, updateRequest.Message, updateRequest.PostID)
+		if err != nil {
+			p.API.LogError("Unable to add issue err=" + err.Error())
+			p.handleErrorWithCode(w, http.StatusInternalServerError, "Unable to add issue", err)
+			return
+		}
+		p.trackUpdateIssue(userID, sourceWebapp, updateRequest.PostID != "")
+		replyMessage := fmt.Sprintf("@%s attached a todo to this thread", senderName)
+		p.postReplyIfNeeded(updateRequest.PostID, replyMessage, updateRequest.Message)
+		return
+	}else if receiver.Id != userID {
+
+
+		issueID, err := p.listManager.SendIssue(userID, receiver.Id, updateRequest.Message, updateRequest.PostID)
```

Comment:
Since I believe we should not send to users a todo when updating, this would not be needed. The question is, do we want to modify just my end of the todo, or do we want to modify both ends? (the one in the sender and the one in the receiver) @aaronrothschild ?

Reply:
I think we should modify BOTH the sent and received text (and indicate it was edited)

Reply:
Disabled the editing of sent todos.
Not sure how to send a refresh event to the `receiver` to update his `my` list